### DISCUSSION TITLE
using only routes

### DISCUSSION THREAD

**Discussion by chahidinho** [mesa/mesa-geo#162] [Q&A] [answered]
For the Epidemics Model how can we make the agents use only routes instead of a random point in the map ?

**Comment by wang-boyu** [comment, marked as answer]
It's definitely possible to use routes for agents' movements. There's an example model called Agents and Networks (source code [here](https://github.com/projectmesa/mesa-examples/tree/main/gis/agents_and_networks)) where each Commuter has its [path](https://github.com/projectmesa/mesa-examples/blob/998ed614d61a2ba30b9e316e80db1d239e10e0a6/gis/agents_and_networks/src/agent/commuter.py#L22-L24) computed between home and work locations (in the [_path_select()](https://github.com/projectmesa/mesa-examples/blob/998ed614d61a2ba30b9e316e80db1d239e10e0a6/gis/agents_and_networks/src/agent/commuter.py#L177-L212) function).

Another way of getting the paths is to do the computation offline somewhere else, save the computed paths (in a dataframe for instance), and load them directly into your model. I don't have an example for this yet.

**Reply by chahidinho** [reply]
Thanks for your Help 👍
